app/javascript/pages/components/DataViewer.jsx

```diff
@@ -0,0 +1,135 @@
+/* eslint-disable max-len */
+/* eslint-disable react/destructuring-assignment */
+/* eslint-disable react/prop-types */
+import React from 'react';
+import axios from 'axios';
+import DatasetHeader from './partials/DatasetHeader';
+import DatasetTable from './partials/DatasetTable';
+
+export default class DataViewer extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      availableYears: [],
+      table: '',
+      schema: '',
+      database: '',
+      queryYearColumn: '',
+      displayHeaders: [],
+      rowHeaders: [],
+      rows: [],
+      title: '',
+      source: '',
+      universe: '',
+      description: '',
+      metadata: {},
+      selectedYears: [],
+      currentPage: 1,
+    };
+    this.updateSelectedYears = this.updateSelectedYears.bind(this);
+    this.updatePage = this.updatePage.bind(this);
+  }
+
+  componentDidMount() {
+    const queryBase = 'https://prql.mapc.org/';
+    const queryToken = '16a2637ee33572e46f5609a578b035dc';
+    this.props.fetchDatasets().then((storeResponse) => {
+      const fullDataset = storeResponse.datasets.filter((dataset) => +dataset.seq_id === +this.props.match.params.id)[0];
+      const table = fullDataset.table_name;
+      const schema = fullDataset.schemaname;
+      const database = fullDataset.db_name;
+      const title = fullDataset.menu3;
+      const source = fullDataset.source;
+      const queryYearColumn = fullDataset.yearcolumn;
+      const yearQuery = axios.get(`${queryBase}?query=select distinct(${queryYearColumn}) from ${schema}.${table} LIMIT 50&token=${queryToken}`);
+      const tableQuery = axios.get(`${queryBase}?query=select * from ${schema}.${table} order by ${queryYearColumn} DESC LIMIT 15000&token=${queryToken}`);
+      const headerQuery = axios.get(`/${database}?tables=${table}`, { headers: { 'Access-Control-Allow-Origin': '*' } });
```

Comment:
Wondering what we needed the `Access-Control-Allow-Origin` header in the get request for? Might be able to handle this by updating the Access control settings in Rails.

Reply:
We don't need it, I forgot it was there. It was one of the troubleshooting steps I tried in getting that endpoint to work (when in reality the solution was to create the endpoint, shoutout to our pal the .ENV file)